src/main/kotlin/minesweeper/domain/Blocks.kt

```diff
@@ -5,15 +5,10 @@ data class Blocks(
 ) {
 
     companion object {
-        fun of(blockCount: Int, mineCount: Int): Blocks {
-            require(blockCount > mineCount) { "지뢰 개수가 전체 블록 개수보다 많을 수 없습니다." }
-
-            return Blocks(
-                (
-                    List(blockCount - mineCount) { CleanBlock() } +
-                        List(mineCount) { MineBlock() }
-                    ).shuffled()
-            )
+        fun of(wholeCount: Int, mineLocation: List<Int>): Blocks {
+            val blocks: MutableList<Block> = MutableList(wholeCount) { CleanBlock() }
+            mineLocation.forEach { blocks[it] = MineBlock() }
+            return Blocks(blocks)
```

Comment:
코드들을 살펴보니, 대부분 생성자에서 Mutable 타입을 사용하고 있네요. (CleanBlock, MineBlock, BlockTable 등)
만약 아래의 코드를 실행하게 된다면, 어떻게 동작하게 될까요?
가급적이면 외부에서는 Immutable타입을, 내부에서는 Mutable 타입을 사용하는 것이 좋아요.
이를 막기 위해서 고민해보시고 개선해보면 어떨까요? (이하 관련 내용 동일)
```suggestion
            val blocks: MutableList<Block> = MutableList(wholeCount) { CleanBlock() }
            mineLocation.forEach { blocks[it] = MineBlock() }
            blocks.add(CleanBlock())
            return Blocks(blocks)
```